brave-core/src/main/java/com/github/kristofa/brave/SpanIdGenerator.java

```diff
@@ -0,0 +1,30 @@
+package com.github.kristofa.brave;
+
+import com.github.kristofa.brave.internal.Nullable;
+
+import java.util.Random;
+
+/**
+ * Allows you to control how ids are generated. For example, implementations
+ * can encode where in the call tree the span originates from, or use alternate
+ * random number generators.
+ */
+// @FunctionalInterface
```

Comment:
I guess adding a note like `// @FunctionalInterface (commented out because of Java 7 support)` or `// @FunctionalInterface (uncomment after dropping Java 7 support)` would help. But it's a minor thing :)